docs/Researcher/scheduling/gpu-memory-swap.md

```diff
@@ -0,0 +1,111 @@
+kubectl patch -n runai runaiconfigs.run.ai/runai --type='merge' --patch '{"spec":{"global":{"core":{"swap":{"enabled": true}}}}}'
+```
+
+To make a workload swappable, a number of conditions must be met:
+
+1. The workload MUST use Dynamic Fractions. This means the workload’s memory request is less than a full GPU, but it may add a GPU memory limit to allow the workload to effectively use the full GPU memory. If regular fractions are used instead of Dynamic Fractions is NOT used but regular fraction is (for that workload), the swap logic assumes this workload prefers NOT to be swapped-out and therefore, all other workloads on the same GPU are NOT swapped either.
+
+2. The administrator must label each node that they want to provide GPU memory swap with a `run.ai/swap-enabled=true` this enables the feature on that node. Enabling the feature creates a local swap file in the CPU to serve the swapped memory from all GPUs on that node. The administrator sets the size of the CPU swap file as a value in the `runaiconfig` file.
+
+3. Optionally configure *Node Level Scheduler*. Using node level scheduler can help in the following ways:
+
+    * The Node Level Scheduler automatically spreads workloads between the different GPUs on a node, ensuring maximum workload performance and GPU utilization.
+    * In scenarios where Interactive notebooks are involved, if the CPU reserved memory for the GPU swap is full, the Node Level Scheduler preempts the GPU process of that workload and potentially routes the workload to another GPU to run.
+
+### Configure `system reserved` GPU Resources
+
+Swappable workloads require reserving a small part of the GPU for non-swappable allocations like binaries and GPU context. To avoid getting out-of-memory (OOM) errors due to non-swappable memory regions, the system reserves a 2GiB of GPU RAM memory by default, effectively truncating the total size of the GPU. For example, a 16GiB T4 will appear as 14GiB on a swap-enabled node.
+The exact reserved size is application-dependent, and 2GiB is a safe assumption for 2-3 applications sharing and swapping on a GPU.
+This value can be changed by editing the `runaiconfig` specification as follows:
+
+```yml
+spec: 
+ global: 
+   core: 
+     swap:
+       limits:
+         reservedGpuRam: 2Gi
+```
+
+You can also use the `patch` command from your terminal:
+
+```bash
+kubectl patch -n runai runaiconfigs.run.ai/runai --type='merge' --patch '{"spec":{"global":{"core":{"swap":{"limits":{"reservedGpuRam": <quantity>}}}}}}'
+```
+
+This configuration is in addition to the *Dynamic Fractions* configuration, and optional *Node Level Scheduler* configuration.
+
+## Preventing your workloads from getting swapped
+
+If you prefer your workloads not to be swapped into CPU memory, you can specify an anti-affinity to `run.ai/swap-enabled=true` node label when submitting your workloads and the Scheduler will ensure not to use swap-enabled nodes.
+
+## What happens when CPU SWAP file is exhausted?
```

Comment:
We've agreed to refrain from using the term `SWAP file` to avoid confusion, right? In the doc we've use:
"What happens when the CPU reserved memory for GPU swap is exhausted?"
Not sure how we'll it'll transalte to the rest of the document